### ISSUE TITLE
Compile and running details

### ISSUE THREAD

**Issue by dejungle** [mhalber/Rescan#1]
I met two problems when running this code with python 3.5:

1. It seems that the dependence 'im3d' is not provided. I found a lib: https://github.com/john-chapman/im3d for im3d. Is this the correct one? 
Besides, when viewing a rsdb file, it failed with 'rsdb_viewer: ~/Rescan/lib/im3d/im3d.cpp:1519: void Im3d::Context::draw(): Assertion `m_appData.drawCallback' failed.'

2. the assertion 'assert( n_scenes == n_arrangements )' in line 478 of rs_database.h failed when tested on the case 'scene_a'.

**Comment by mhalber** [comment]
Hi, thanks for pointing out these issues. 

The im3d you point out is a correct library, I will make sure to include it in a new commit.

Would you mind sharing the exact commands you're running on for both of the assertions such that I can reproduce and fix these?

Thanks again!

**Comment by dejungle** [comment]
Thanks for your reply.

In the root directory of the rescan dataset, I executed the following command, where 'scene_list.txt' only contains  the test case 'scene_a':
xxx@xxx:~/data/rescan_dataset$ python3 ../../code/Rescan/scripts/run_segmentation_pipeline.py scene_list.txt ../../code/Rescan/bin/ ../../code/Rescan/scripts/

the process was terminated when trying to execute '../../code/Rescan/bin/segment_transfer scene_a/scene_a_1_pp.rsdb -o scene_a/scene_a_1.rsdb -v' with the following errors:
IO: Loading the database...
segment_transfer: ~/code/Rescan/apps/segment_transfer/../../lib/rs/rs_database.h:478: int rsdb_load(rsdb_t*, const char*, int): Assertion `n_scenes == n_arrangements' failed.
Aborted (core dumped)
Traceback (most recent call last):
  File "../../code/Rescan/scripts/run_segmentation_pipeline.py", line 78, in <module>
    main()
  File "../../code/Rescan/scripts/run_segmentation_pipeline.py", line 75, in main
    common.run_cmds(cmds)
  File "/home/cscg/mutj/code/Rescan/scripts/common.py", line 35, in run_cmds
    subprocess.run( cmd, shell=True, check=True )
  File "/usr/lib/python3.5/subprocess.py", line 708, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '../../code/Rescan/bin/segment_transfer scene_a/scene_a_1_pp.rsdb -o scene_a/scene_a_1.rsdb -v' returned non-zero exit status 134.

It seems that the segment_transfer failed to load the 'scene_a/scene_a_1_pp.rsdb' saved in previous step. I check 'scene_a/scene_a_1_pp.rsdb' and indeed it contains 2 scenes and 1 arrangement.

As to the rsdb_viewer,  I run the command  '../../code/Rescan/bin/rsdb_viewer scene_a/scene_a_0.rsdb'  to view a rsdb file, it failed with the following hint:
rsdb_viewer: ~/Rescan/lib/im3d/im3d.cpp:1519: void Im3d::Context::draw(): Assertion `m_appData.drawCallback' failed.

**Comment by mhalber** [comment]
Hi, thank you for this extra information

Looking back at the code, the Im3d calls should not be there in the first place - I was using im3d for some debug visualization during development. I removed all the calls and dependencies in CMakeLists.txt.

The assertion was in fact a bogus, and that condition should not be checked - this has also been fixed and tested.

Let me know if this solves your issues.
